test/OssifiableProxy.t.sol

```diff
@@ -0,0 +1,168 @@
+        vm.prank(stranger);
+        vm.expectRevert(OssifiableProxy.NotAdmin.selector);
+        proxy.proxy__changeAdmin(stranger);
+    }
+
+    function test_changeAdmin_RevertWhenOssified() public {
+        vm.prank(admin);
+        proxy.proxy__ossify();
+        assertTrue(proxy.proxy__getIsOssified());
+
+        vm.expectRevert(OssifiableProxy.ProxyIsOssified.selector);
+        proxy.proxy__changeAdmin(stranger);
+    }
+
+    function test_changeAdmin() public {
+        vm.prank(admin);
+        proxy.proxy__changeAdmin(stranger);
+        assertEq(proxy.proxy__getAdmin(), stranger);
+    }
+
+    function test_upgradeTo_RevertWhenNotAdmin() public {
+        vm.prank(stranger);
+        vm.expectRevert(OssifiableProxy.NotAdmin.selector);
+        proxy.proxy__upgradeTo(address(nextImpl));
+    }
+
+    function test_upgradeTo_RevertWhenOssified() public {
+        vm.prank(admin);
+        proxy.proxy__ossify();
+        assertTrue(proxy.proxy__getIsOssified());
+
+        vm.expectRevert(OssifiableProxy.ProxyIsOssified.selector);
+        proxy.proxy__upgradeTo(address(nextImpl));
+    }
+
+    function test_upgradeTo() public {
+        vm.prank(admin);
+        proxy.proxy__upgradeTo(address(nextImpl));
+        assertEq(proxy.proxy__getImplementation(), address(nextImpl));
+    }
+
+    function test_upgradeToAndCall_RevertWhenNotAdmin() public {
+        vm.prank(stranger);
+        vm.expectRevert(OssifiableProxy.NotAdmin.selector);
+        proxy.proxy__upgradeToAndCall(
+            address(nextImpl),
+            abi.encodeWithSelector(nextImpl.initialize.selector, 1)
+        );
+    }
+
+    function test_upgradeToAndCall_RevertWhenOssified() public {
+        vm.prank(admin);
+        proxy.proxy__ossify();
+        assertTrue(proxy.proxy__getIsOssified());
+
+        vm.expectRevert(OssifiableProxy.ProxyIsOssified.selector);
+        proxy.proxy__upgradeToAndCall(
+            address(nextImpl),
+            abi.encodeWithSelector(nextImpl.initialize.selector, 1)
+        );
+    }
+
+    function test_upgradeToAndCall() public {
+        vm.prank(admin);
+        proxy.proxy__upgradeToAndCall(
+            address(nextImpl),
+            abi.encodeWithSelector(nextImpl.initialize.selector, 1)
+        );
+        assertEq(proxy.proxy__getImplementation(), address(nextImpl));
+        assertEq(
+            InitializableImplementationStub(payable(address(proxy))).version(),
+            1
+        );
+    }
+
+    function test_receive() public {
+        vm.deal(admin, 2 ether);
+        vm.prank(admin);
+        vm.expectEmit(true, true, true, true, address(proxy));
+        emit InitializableImplementationStub.FallbackIsFired();
+        payable(address(proxy)).call{ value: 1 ether }("");
+        assertEq(address(proxy).balance, 1 ether);
+    }
+
+    function test_fallback() public {
+        vm.prank(admin);
+        (, bytes memory data) = address(proxy).call(
```

Comment:
Why no wrap the address and call the method?

Reply:
It's the fallback of the proxy is being checked, so it will be fired if we call it via the interface.

Reply:
more straightforward that it's "call" check but there is no actual difference, right